dotnet/src/SemanticKernel/Planning/Plan.cs

```diff
@@ -252,7 +252,7 @@ public async Task<Plan> InvokeNextStepAsync(SKContext context, CancellationToken
             var functionVariables = this.GetNextStepVariables(context.Variables, step);
 
             // Execute the step
-            var functionContext = new SKContext(functionVariables, context.Skills, context.LoggerFactory);
+            var functionContext = new SKContext(context.Kernel, functionVariables, context.Skills);
```

Comment:
We don't have coverage for it right now, but curious about how this will integrate with the ability to do this
```csharp
// SequentialPlannerConfig.cs
public Func<string, string, ISKFunction?>? GetSkillFunction { get; set; 
```

Reply:
The path moving forward would be something like:

```
context.Kernel.RunAsync(GetSkillFunction(plugin, name))
```